projects/element-ng/resize-observer/resize-observer.service.ts

```diff
@@ -118,27 +129,38 @@ export class ResizeObserverService {
     subscriber.sub.complete();
   }
 
-  private handleElement(element: Element): void {
+  private handleElement(event: ResizeEvent): void {
+    const element = event.target;
     const entry = this.listeners.get(element);
     if (!entry) {
       this.resizeObserver?.unobserve(element);
       return;
     }
-    entry.subscribers.forEach(sub => this.handleResizeSubscriber(element, sub));
+    entry.subscribers.forEach(sub => this.handleResizeSubscriber(event, sub));
   }
 
-  private handleResizeSubscriber(element: Element, entry: ResizeSubscriber): void {
+  private handleResizeSubscriber(event: ResizeEvent, entry: ResizeSubscriber): void {
     if (entry.blocked) {
       return;
     }
+    const element = event.target;
+    if (entry.dim) {
+      entry.dim.borderBoxSize = event.borderBoxSize?.at(0);
+      entry.dim.contentBoxSize = event.contentBoxSize?.at(0);
```

Comment:
Did you test that you're getting a proper value here? I thought @dr-itz said you can only monitor one (border-box or content-box). See also the `ResizeObserver.observe()` parameters where you can specify the option: 
`type ResizeObserverBoxOptions = "border-box" | "content-box" | "device-pixel-content-box";`

Reply:
And on top, I think we said in the internal discussion, it would also not make sense to monitor both for the same element.

Reply:
monitor for what the consumer is asking, i.e. border-box by default.

- create two ResizeObserver, one for border-box, one for content-box
- handling between those is independent
- register with the one that is needed based on consumer option

i.e. I would move most of the current service to a helper class and have two instances of those around, i.e. make the current service a thin mux layer